rmf_site_editor/src/interaction/camera_controls/cursor.rs

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+use super::{
+    get_groundplane_else_default_selection, CameraCommandType, CameraControls, ProjectionMode,
+    MAX_FOV, MAX_PITCH, MAX_SCALE, MIN_FOV, MIN_SCALE,
+};
+use crate::interaction::SiteRaycastSet;
+use bevy::input::mouse::{MouseMotion, MouseWheel};
+use bevy::prelude::*;
+use bevy::window::PrimaryWindow;
+use bevy_mod_raycast::deferred::RaycastSource;
+use nalgebra::{Matrix3, Matrix3x1};
+
+pub const SCALE_ZOOM_SENSITIVITY: f32 = 0.1;
+pub const TRANSLATION_ZOOM_SENSITIVITY: f32 = 0.2;
+pub const FOV_ZOOM_SENSITIVITY: f32 = 0.1;
+
+#[derive(Resource)]
+pub struct CursorCommand {
+    pub translation_delta: Vec3,
+    pub rotation_delta: Quat,
+    pub scale_delta: f32,
+    pub fov_delta: f32,
+    pub cursor_selection: Option<Vec3>,
+    pub cursor_direction_camera_frame: Option<Vec3>,
+    pub command_type: CameraCommandType,
+}
+
+impl Default for CursorCommand {
+    fn default() -> Self {
+        Self {
+            translation_delta: Vec3::ZERO,
+            rotation_delta: Quat::IDENTITY,
+            scale_delta: 0.0,
+            fov_delta: 0.0,
+            cursor_selection: None,
+            cursor_direction_camera_frame: None,
+            command_type: CameraCommandType::Inactive,
+        }
+    }
+}
+
+pub fn update_cursor_command(
+    mut camera_controls: ResMut<CameraControls>,
+    mut cursor_command: ResMut<CursorCommand>,
+    mut mouse_motion: EventReader<MouseMotion>,
+    mut mouse_wheel: EventReader<MouseWheel>,
+    mouse_input: Res<Input<MouseButton>>,
+    keyboard_input: Res<Input<KeyCode>>,
+    raycast_sources: Query<&RaycastSource<SiteRaycastSet>>,
+    cameras: Query<(&Projection, &Transform, &GlobalTransform)>,
+    primary_windows: Query<&Window, With<PrimaryWindow>>,
+) {
+    if let Ok(window) = primary_windows.get_single() {
+        // Return if cursor not within window
+        if window.cursor_position().is_none() {
+            *cursor_command = CursorCommand::default();
```

Comment:
This line causes a jarring drop out of the control mode if the cursor ever leaves the screen space.

I've opened a PR targeting your fork with a fix that I would recommend: https://github.com/reuben-thomas/rmf_site/pull/1